### PR TITLE
Allow group displaynames in the database backend

### PR DESCRIPTION
**Pull request by nickvergessen**

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]
Code looks good!

File: lib/private/Group/Database.php [hunk 1]
```diff
@@ -414,4 +420,56 @@ public function countDisabledInGroup(string $gid): int {
 		return $count;
 	}
 
+	/**
+	 * @param string $gid
+	 * @return string
+	 * @since 17.0.0
```
Comment (kesselb) [inline lib/private/Group/Database.php]:
```suggestion
	 * @since 18.0.0
```

File: lib/private/Group/Database.php [hunk 2]
```diff
@@ -414,4 +420,56 @@ public function countDisabledInGroup(string $gid): int {
...
+		$displayName = $result->fetchColumn();
+		$result->closeCursor();
+
+		return (string) $displayName;
+	}
+
+	/**
+	 * @param string $gid
+	 * @return array
+	 * @since 14.0.0
```
Comment (kesselb) [inline lib/private/Group/Database.php]:
```suggestion
	 * @since 18.0.0
```

File: apps/provisioning_api/lib/Controller/GroupsController.php [hunk 3]
```diff
@@ -248,6 +248,28 @@ public function addGroup(string $groupid): DataResponse {
...
+	public function updateGroup(string $groupId, string $key, string $value): DataResponse {
+		if ($key === 'displayname') {
+			$group = $this->groupManager->get($groupId);
+			if ($group->setDisplayName($value)) {
+				return new DataResponse();
+			}
+
+			throw new OCSException('Not supported by backend', 101);
+		} else {
+			throw new OCSException('', \OCP\API::RESPOND_UNAUTHORISED);
```
Comment (kesselb) [inline apps/provisioning_api/lib/Controller/GroupsController.php]: I think unauthorised is a bit missleading here.

**Comment by kesselb** [comment]
Does this fix https://github.com/nextcloud/server/issues/9209?

**Comment by skjnldsv** [comment]
> Does this fix #9209?

It will miss the ui, so not totally!

File: lib/private/Group/Database.php (hunk 2 above)
Comment (nickvergessen) [reply lib/private/Group/Database.php]: Well its inherited from the interface of the method.

File: apps/provisioning_api/lib/Controller/GroupsController.php (hunk 3 above)
Comment (nickvergessen) [reply apps/provisioning_api/lib/Controller/GroupsController.php]: I used what the UserController does. I think it's good to stay in line

File: lib/private/Group/Database.php (hunk 2 above)
Comment (blizzz) [reply lib/private/Group/Database.php]: the whole block can left it out in the implementation

**Review by blizzz — APPROVED** [review]
:elephant:

File: apps/provisioning_api/lib/Controller/GroupsController.php (hunk 3 above)
Comment (kesselb) [reply apps/provisioning_api/lib/Controller/GroupsController.php]: Sure.

File: lib/private/Group/Database.php (hunk 2 above)
Comment (kesselb) [reply lib/private/Group/Database.php]: >  inherited from the interface

:see_no_evil:

Comment (nickvergessen) [reply lib/private/Group/Database.php]: Removed

**Comment by rullzer** [comment]
CI does :boom:

**Comment by nickvergessen** [comment]
Rebased and fixed the magic with the empty displayname of the admin group breaking everything

**Comment by rullzer** [comment]
soooo good to go?

**Comment by nickvergessen** [comment]
Yes

**Comment by ChristophWurst** [comment]
so … why didn't you bump the version? Pulling latest master broke my dev env as the migration isn't picked up automatically :)

**Pull request merged**